WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivityHelper.kt

```diff
@@ -0,0 +1,66 @@
+package org.wordpress.android.ui.posts
+
+import android.content.Context
+import org.wordpress.android.editor.gutenberg.GutenbergPropsBuilder
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.EditorThemeStore
+import org.wordpress.android.util.LocaleManager
+import org.wordpress.android.util.SiteUtils
+import org.wordpress.android.util.WPMediaUtils
+import java.util.Locale
+
+// Until EditPostActivity gets converted to Kotlin, this file is for those parts of EditPostActivity
+// that would really benefit from being written in Kotlin now.
+internal object EditPostActivityHelper {
+    fun getGutenbergPropsBuilder(
+        site: SiteModel,
+        isPage: Boolean,
+        isXPostsCapable: Boolean?,
+        isJetpackSsoEnabled: Boolean,
+        editPostRepository: EditPostRepository,
+        editorThemeStore: EditorThemeStore,
+        context: Context
+    ): GutenbergPropsBuilder {
```

Comment:
:wave: @ParaskP7 ! This method is giving me a detekt violation since it has 7 parameters (1 more than the max of 6). I don't disagree that this is a good rule, but I'm wondering if you have any suggestions for how to address this because I kind of feel like ignoring this specific violation might be the best way to proceed. 

I'm creating this method in order to avoid passing almost 20 _unnamed_ parameters within Java. Many of those parameters are simple booleans, so I'm concerned that there is a pretty high risk of getting two of them mixed up.

I considered creating a proper Java builder object for Java to use to construct the `GutenbergPropsBuilder`, but that's quite a bit more code to maintain (could use something like https://github.com/ThinkingLogic/kotlin-builder-annotation, but I'm trying to avoid the temptation to over-engineer more than I already am 😉 ), and I don't see how that would give us any advantages over just relying on Kotlin's named parameters like I am here. Basically, I'm just curious if you have any thoughts about how we might go about improving this or approaching it differently. Thanks so much! 🙇

Reply:
👋 @mchowning and thank you for the ping, apologies for keeping you waiting on my reply! 🙈 

> ! This method is giving me a detekt violation since it has 7 parameters (1 more than the max of 6). I don't disagree that this is a good rule, but I'm wondering if you have any suggestions for how to address this because I kind of feel like ignoring this specific violation might be the best way to proceed.

The main purpose of this kind of rules is to help us identify that there might be a potential issue so that to trigger us thinking about it in order for us to (a) try and solve it or (b) suppress it anyway because this might be an exception to the rule and we really believe that this is the best way to proceed anyway.

As such, it seems that you did all the above and end up thinking that it is best to ignore it. I don't see any harm in that and indeed in lots of situations like that ignoring is totally valid.

> I'm creating this method in order to avoid passing almost 20 unnamed parameters within Java. Many of those parameters are simple booleans, so I'm concerned that there is a pretty high risk of getting two of them mixed up.

👍 

> I considered creating a proper Java builder object for Java to use to construct the GutenbergPropsBuilder, but that's quite a bit more code to maintain (could use something like https://github.com/ThinkingLogic/kotlin-builder-annotation, but I'm trying to avoid the temptation to over-engineer more than I already am 😉 ), and I don't see how that would give us any advantages over just relying on Kotlin's named parameters like I am here. Basically, I'm just curious if you have any thoughts about how we might go about improving this or approaching it differently. Thanks so much! 🙇

I agree and thank you so much for thinking about it so deeply. I too believe that with Kotlin coming into play the builder pattern goes away... that rhymes! 😝 So, all in all, I would also avoid the temptation of over-engineering this. 💯

Reply:
> with Kotlin coming into play the builder pattern goes away.... that rhymes!

I'm certainly not going to argue against anything that rhymes! 😄  I've updated the PR to ignore that lint violation. Thanks so much for sharing your thoughts Petros!

Reply:
> Well I'm certainly not going to argue against anything that rhymes so well! 😄

🤣 

> I've updated the PR to ignore that lint violation. Thanks so much for sharing your thoughts Petros!

🥇